### ISSUE TITLE
Invalid Product on Inventory when Product is confirmed in CCRS

### ISSUE THREAD

**Issue by djbusby** [openthc/lcb-ccrs#68]
Trying to import Inventory Item '01H0NBNPWAT6R254'

It fails with a rejection from CCRS: "Invalid Product"

The Product is "WAJ430957.TYKTMWF" with the name="flower lot"

This Product definition has been shipped to CCRS at least 10 times.

On INSERT the Product upload we get the error "Duplicate External Identifier"
This error is expected, it means the product definition does exist in CCRS.

On UPDATE no error is received; which indicates the UPDATE is successful.

This confirms to us that this product is in CCRS and does have the name we've requested: "flower lot"

When we upload inventory with this Product -- we get a failure: "Invalid Product"

----

The core issue is that CCRS does not link by Primary Identifiers
I'm 100% certain this issue is related to name based lookups.
Since at least 1976 it's been a known good idea to use Primary Identifiers on these mapptings.
For example, uploading a Section by ID and Name, then a Product by ID and Name.
Then when uploading Inventory rather than Name-based columns, one could use ID-based.
This ancient method removes very large classes of bugs -- which this issue very much looks like.

**Comment by djbusby** [comment]
Should be noted that this issue is affecting 1590 records across 28 licensees.  Details in internal document; emailed to LCB.

**Comment by djbusby** [comment]
IR80288

**Comment by Panpsychism** [comment]
Do you have this list of affected licensees?

**Comment by djbusby** [comment]
This is not a list we are able to share publicly; we have provided it to the LCB.  The list only included licensees on our hosted/managed plaform.  We have no visibility to those with self-hosted OpenTHC or those on other platforms.

Additionally, the work-around is to make an attempt to remove the errant Product records via DELETE to CCRS. Then re-create Product definitions w/new IDs; remap your local database and upload based on that.

**Comment by Panpsychism** [comment]
Makes sense. Can you say whether it is likely that the same issue affected
other licensees using other third party integrators?

**Comment by djbusby** [comment]
Appear to be observing this issue again.  

- PRODUCT UPLOAD 01HWE0BFEPCMQYE4BQ3C3F1YHN
- PRODUCT UPLOAD 01HWE0D5BF2567FFPJNHNK476F
- INVENTORY UPLOAD 01HWE11JF76VERFC4YEEV1MFSG

IR95087
